### PR TITLE
Avoid using variable-length array

### PR DESCRIPTION
**Pull request by ericlinsechs**
Using variable-length array is not a good call
([https://lkml.org/lkml/2011/10/23/25](https://lkml.org/lkml/2011/10/23/25)). So remove it.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Check https://cbea.ms/git-commit/ carefully and refine the git commit message.

Hyperlinks should appear at the end of git commit message.

**Comment by ericlinsechs** [comment]
I have corrected the commit message. Could you please review it? Thank you.

**Comment by jserv** [comment]
> I have corrected the commit message. Could you please review it? Thank you.

You don't have to leave the above message. Simply click "Re-request review" via GitHub web page.
See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review

**Review by jserv — CHANGES_REQUESTED** [review]
Don't place "Link: https://lkml.org/lkml/2011/10/23/25", which does not express the rationale. You can quote some insights from the LKML.

File: client.c [hunk 1]
```diff
@@ -29,6 +29,10 @@ int main()
     for (int i = 0; i <= offset; i++) {
         lseek(fd, i, SEEK_SET);
         sz = read(fd, buf, 1);
+        if (sz < 0) {
```
Comment (jserv) [inline client.c]: This should appear in another pull request. Let's stick to the proposed subject for necessary changes.

File: client.c [hunk 2]
```diff
@@ -38,6 +42,10 @@ int main()
     for (int i = offset; i >= 0; i--) {
         lseek(fd, i, SEEK_SET);
         sz = read(fd, buf, 1);
+        if (sz < 0) {
```
Comment (jserv) [inline client.c]: Ditto. Decouple from this commit.

File: fibdrv.c [hunk 3]
```diff
@@ -27,8 +28,9 @@ static int major = 0, minor = 0;
 
 static long long fib_sequence(long long k)
 {
-    /* FIXME: C99 variable-length array (VLA) is not allowed in Linux kernel. */
-    long long f[k + 2];
+    long long *f = kmalloc(sizeof(*f) * (k + 2), GFP_KERNEL);
+    if (!f)
+        return -1;
```
Comment (jserv) [inline fibdrv.c]: The return value `-1` should be documented, otherwise it is nonsensical as for the math definition.

Comment (ericlinsechs) [reply fibdrv.c]: I supposed `errno` would be appropriate to describe the situation?

`   return -ENOMEM;`

File: fibdrv.c [hunk 4]
```diff
@@ -27,8 +28,9 @@ static int major = 0, minor = 0;
 
 static long long fib_sequence(long long k)
 {
-    /* FIXME: C99 variable-length array (VLA) is not allowed in Linux kernel. */
-    long long f[k + 2];
+    long long *f = kmalloc(sizeof(*f) * (k + 2), GFP_KERNEL);
+    if (!f)
+        return -ENOMEM;
```
Comment (jserv) [inline fibdrv.c:40]: Provide explicit comments for this function.

**Comment by jserv** [comment]
Thank @ericlinsechs for contributing!

**Pull request merged**